lib/beaker/options/parser.rb

```diff
@@ -67,29 +60,39 @@ def split_arg arg
       #                         no .rb files are found overall
       def file_list(paths)
         files = []
-        if not paths.empty?
+        unless paths.empty?
           paths.each do |root|
+            @validator.validate_path(root)
+
+            path_files = []
             if File.file?(root)
-              files << root
+              path_files << root
             elsif File.directory?(root) #expand and explore
-              discover_files = Dir.glob(
-                File.join(root, "**/*.rb")
-              ).select { |f| File.file?(f) }
-              if discover_files.empty?
-                parser_error "empty directory used as an option (#{root})!"
-              end
-              files += discover_files.sort_by {|file| [file.count("/"), file]}
-            else #not a file, not a directory, not nothin'
-              parser_error "#{root} used as a file option but is not a file or directory!"
+              path_files = Dir.glob(File.join(root, '**/*.rb'))
+                               .select { |f| File.file?(f) }
+                               .sort_by { |file| [file.count('/'), file] }
             end
+
+            @validator.validate_files(path_files, root)
+            files += path_files
           end
         end
-        if files.empty?
-          parser_error "no .rb files found in #{paths.to_s}"
-        end
+
+        @validator.validate_files(files, paths.to_s)
```

Comment:
The old code for this method (`file_list`) raises errors when:

* an empty directory is provided (`@validator.validate_files(path_files,root)`)
* a path is not a file or directory (`@validator.validate_path(root)`)
* if the final `files` array is empty (`@validator.validate_files(files, paths.to_s)`)

This was an attempt to make the code easier to read, move errors to `Validator`, while trying to retain the detailed error messages.